base/floatfuncs.jl

```diff
@@ -105,24 +105,30 @@ end
 
 function _signif_og(x, digits, base)
     if base == 10
-        oftype(x, 10. ^ floor(log10(abs(x)) - digits + 1.))
+        e = floor(log10(abs(x)) - digits + 1.)
+        og = oftype(x, exp10(abs(e)))
     elseif base == 2
-        oftype(x, 2. ^ floor(log2(abs(x)) - digits + 1.))
+        e = exponent(abs(x)) - digits + 1.
```

Comment:
Shouldn't the addition of the Float literal promote this to `Float64`/`Float32` anyways so that `e` always has the same type?
Sure, I'm adding two integers and a float, but that shouldn't harm performance any more than doing the conversion manually, right?